### ISSUE TITLE
Allow text/plain emails

### ISSUE THREAD

**Issue by awaragi** [prometheus/alertmanager#926]
Currently the email notifier is hard coded for html emails only.  Considering that support for text/plain can be achieved by checking if a Content-Type header is already specified or not, is that a good solution that I can create a pull request for? or is such support is completely out of question?

Snippet + pseudo code from impl.go

```
	for header, t := range n.conf.Headers {
		value, err := n.tmpl.ExecuteTextString(t, data)
		if err != nil {
			return false, fmt.Errorf("executing %q header template: %s", header, err)
		}
		fmt.Fprintf(wc, "%s: %s\r\n", header, mime.QEncoding.Encode("utf-8", value))
	}

     // CHECK IF CONTENT IS ALREADY SPECIFIED BEFORE FORCING HTML
     if(n.conf.Headers["Content-Type"] == nil) {
	fmt.Fprintf(wc, "Content-Type: text/html; charset=UTF-8\r\n")
     }
	fmt.Fprintf(wc, "Date: %s\r\n", time.Now().Format(time.RFC1123Z))
```

**Comment by brian-brazil** [comment]
There's more to this, as the templating applied is for HTML.

**Comment by awaragi** [comment]
I understand. My code simply means that if the Content-Type is specified, don't hardcode it as in "I know what I am doing". The HTML template is evaluate as normal but its output is not interpreted as HTML by the email client which is what my purpose is.

**Comment by brian-brazil** [comment]
It'll be escaped as though it is HTML though, which is not what you want.

**Comment by awaragi** [comment]
I tried it and plain new lines were outputted as they are. My text "template" came out correctly but the text/html content-type was forcing the email client to read it as html so the new lines were ignored but the source was correct.

**Comment by awaragi** [comment]
If you are okay with the following logic, I will submit a pull request.

> if the user sets a custom header for email notification of type Content-Type, assume the user knows what they are doing and skip setting Content-Type: text/html to avoid conflict. 

This patch will not break existing systems nor will make any assumptions on the content of the email template.

**Comment by matthiasr** [comment]
@brian-brazil On what basis should the correct templating be selected? How do you want text emails to be selected and templated?

**Comment by brian-brazil** [comment]
Either we'd have separate text and html fields (which I believe we've had requests for in the past), or a bool to select whether it's text or html.

**Comment by awaragi** [comment]
Fair enough. 

What's the consensus if I can provide some coding time? I am personally for two fields as they are not mutually exclusive and actually I've always included both in any application I develop with email notifications.

So basically if HTML input is provided, it will be used. If TEXT input is provided it will be also included. Default behavior is maintained with HTML templates.

**Comment by matthiasr** [comment]
I also think it should be two fields. they are not mutually exclusive.

**Comment by doctorlector** [comment]
Hi All,

I still cannot realize how I can apply my own custom html template for e-mail alert in alert-manager config. Currently, I receive an empty e-mail alert.

Can you help me with that?

My current config alert-manager config:
    html: '{{ define "email.tmpl" }} {{ end }}
           {{ template "__email.tmpl" . }}'

Thanks in advance.

**Comment by mxinden** [comment]
@doctorlector Please redirect your question to https://groups.google.com/forum/#!forum/prometheus-users if it is a usage question. In case you think you found a bug in Alertmanager, please open up a new issue on this repository. We are happy to help you in either one. Thanks.

**Comment by awaragi** [comment]
I took a stab at adding text email notification support. I never developed with Go before so I kept my changes as simple as possible. I couldn't find any tests to update nor documentation in README.md so your online documentation need to be updated by someone else if the pull request is accepted.

The pull request is https://github.com/prometheus/alertmanager/pull/934 . Looking forward to any feedback.
